project-base/storefront/components/Pages/BrandDetail/helpers.ts

```diff
@@ -0,0 +1,113 @@
+
+    const urlSlug = getSlugFromUrl(getUrlWithoutGetParameters(asPath));
+    const mappedFilter = mapParametersFilter(filter);
+
+    const [brandProductsData, setBrandProductsData] = useState<undefined | BrandProductsQueryApi>(
+        readBrandProductsFromCache(
+            client,
+            urlSlug,
+            sort,
+            mappedFilter,
+            getEndCursor(currentPage),
+            DEFAULT_PAGE_SIZE * (currentLoadMore + 1),
+        ),
+    );
+    const [fetching, setFetching] = useState(false);
+    const [loadMoreFetching, setLoadMoreFetching] = useState(false);
+
+    useEffect(() => {
+        if (previousLoadMoreRef.current === currentLoadMore || currentLoadMore === 0) {
+            setFetching(true);
+        } else {
+            setLoadMoreFetching(true);
+            previousLoadMoreRef.current = currentLoadMore;
+        }
+
+        const previouslyQueriedProductsFromCache = getPreviouslyQueriedProductsFromCache(
+            client,
+            urlSlug,
+            sort,
+            mappedFilter,
+            DEFAULT_PAGE_SIZE,
+            currentPage,
+            currentLoadMore,
+            readBrandProductsFromCache,
+        );
+
+        const pageSize = previouslyQueriedProductsFromCache
+            ? DEFAULT_PAGE_SIZE
+            : DEFAULT_PAGE_SIZE * (currentLoadMore + 1);
+        const isMoreThanOnePage = pageSize > DEFAULT_PAGE_SIZE;
+        const endCursor = isMoreThanOnePage ? getEndCursor(currentPage) : getEndCursor(currentPage, currentLoadMore);
+
+        client
+            .query<BrandProductsQueryApi, BrandProductsQueryVariablesApi>(BrandProductsQueryDocumentApi, {
+                endCursor,
+                filter: mappedFilter,
+                orderingMode: sort,
+                urlSlug,
+                pageSize,
+            })
+            .toPromise()
+            .then((response) => {
+                if (!response.data) {
+                    setBrandProductsData(undefined);
+
+                    return;
+                }
+
+                setBrandProductsData(
+                    mergeNewProductsWithPreviouslyQueried(response.data, previouslyQueriedProductsFromCache),
+                );
+            })
+            .finally(() => {
+                setFetching(false);
+                setLoadMoreFetching(false);
+            });
+    }, [urlSlug, sort, JSON.stringify(filter), currentPage, currentLoadMore]);
+
+    return [brandProductsData, fetching, loadMoreFetching];
+};
+
+const readBrandProductsFromCache = (
+    client: Client,
+    urlSlug: string,
+    orderingMode: ProductOrderingModeEnumApi | null,
+    filter: Maybe<ProductFilterApi>,
+    endCursor: string,
+    pageSize: number,
+) =>
+    client.readQuery<BrandProductsQueryApi, BrandProductsQueryVariablesApi>(BrandProductsQueryDocumentApi, {
+        urlSlug,
+        orderingMode,
+        filter,
+        endCursor,
+        pageSize,
+    })?.data ?? undefined;
```

Comment:
```suggestion
    })?.data || undefined;
```

More of philosophy than practical benefit, but if you would be deciding whether to use `??` or `||` the way I take it is that you don't want to let pass values such as empty string or boolean, in such a cases you rather want to return `undefined` so better to use `||`. You decide.

Reply:
No problem :)